ee/hogai/graph/session_summaries/nodes.py

```diff
@@ -219,14 +284,22 @@ async def arun(self, state: AssistantState, config: RunnableConfig) -> PartialAs
                 )
                 summaries_content = await self._summarize_sessions_individually(session_ids=session_ids, writer=writer)
             else:
+                # Check if the notebook is provided, create a notebook to fill if not
+                if not state.notebook_id:
+                    notebook = await create_empty_notebook_for_summary(user=self._user, team=self._team)
+                    # Could be moved to a separate "create notebook" node (or reuse the one from deep research)
+                    state.notebook_id = notebook.short_id
```

Comment:
When adding more nodes - I could move notebook creation to a separate node (as Deep research does), but it seems like overkill for now/.